pkg/projects/resource_project_user_test.go

```diff
@@ -0,0 +1,171 @@
+
+	user := ArtifactoryUser{
+		Email:    email,
+		Password: "Password1",
+		Admin:    false,
+	}
+
+	_, err := restyClient.R().SetBody(user).Put(fmt.Sprintf("/artifactory/api/security/users/%s", username))
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func deleteTestUser(t *testing.T, projectKey string, username string) {
+	restyClient := getTestResty(t)
+
+	_, err := restyClient.R().Delete(fmt.Sprintf("/artifactory/api/security/users/%s", username))
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestAccProjectUser(t *testing.T) {
+	name := fmt.Sprintf("tftestprojects%s", randSeq(10))
+	resourceName := fmt.Sprintf("project_project.%s", name)
+	projectKey := strings.ToLower(randSeq(6))
+
+	username1 := "user1"
+	email1 := fmt.Sprintf("%s@tempurl.org", username1)
+	username2 := "user2"
+	email2 := fmt.Sprintf("%s@tempurl.org", username2)
+	developeRole := "developer"
+	contributorRole := "contributor"
+
+	params := map[string]interface{}{
+		"name":            name,
+		"project_key":     projectKey,
+		"username1":       username1,
+		"username2":       username2,
+		"developeRole":    developeRole,
+		"contributorRole": contributorRole,
+	}
+
+	initialConfig := executeTemplate("TestAccProjectUser", `
+		resource "project_project" "{{ .name }}" {
+			key = "{{ .project_key }}"
+			display_name = "{{ .name }}"
+			description = "test description"
+			admin_privileges {
+				manage_members = true
+				manage_resources = true
+				index_resources = true
+			}
+
+			user {
+				name = "{{ .username1 }}"
+				roles = ["{{ .developeRole }}"]
+			}
+		}
+	`, params)
+
+	addUserConfig := executeTemplate("TestAccProjectUser", `
+		resource "project_project" "{{ .name }}" {
+			key = "{{ .project_key }}"
+			display_name = "{{ .name }}"
+			description = "test description"
+			admin_privileges {
+				manage_members = true
+				manage_resources = true
+				index_resources = true
+			}
+
+			user {
+				name = "{{ .username1 }}"
+				roles = ["{{ .developeRole }}", "{{ .contributorRole }}"]
+			}
+
+			user {
+				name = "{{ .username2 }}"
+				roles = ["{{ .contributorRole }}"]
+			}
+		}
+	`, params)
+
+	noUserConfig := executeTemplate("TestAccProjectUser", `
+		resource "project_project" "{{ .name }}" {
+			key = "{{ .project_key }}"
+			display_name = "{{ .name }}"
+			description = "test description"
+			admin_privileges {
+				manage_members = true
+				manage_resources = true
+				index_resources = true
+			}
+		}
+	`, params)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheck(t)
+			createTestUser(t, projectKey, username1, email1)
+			createTestUser(t, projectKey, username2, email2)
+		},
+		CheckDestroy: verifyDeleted(resourceName, func(id string, request *resty.Request) (*resty.Response, error) {
+			deleteTestUser(t, projectKey, username1)
+			deleteTestUser(t, projectKey, username2)
+			resp, err := verifyProject(id, request)
+
+			return resp, err
+		}),
+		ProviderFactories: testAccProviders,
```

Comment:
Yeah, about this: I've come to the conclusion that sharing the provider doesn't (I don't believe) align with how TF manages providers - I think each test should get a clean provider because TF will fork a process for every resource managed (not 100% sure, but observation suggests so) - and this will make your life easier